src/lib.rs

```diff
@@ -443,12 +443,14 @@ impl Config {
     fn targetted_env_var(&self, var_base: &str) -> Option<OsString> {
         match (env::var("TARGET"), env::var("HOST")) {
             (Ok(target), Ok(host)) => {
-                let kind = if host == target { "HOST" } else { "TARGET" };
+                let kind_suffix = if host == target { "_FOR_BUILD" } else { "" };
+                let kind_prefix = if host == target { "HOST" } else { "TARGET" };
                 let target_u = target.replace("-", "_");
 
                 self.env_var_os(&format!("{}_{}", var_base, target))
                     .or_else(|| self.env_var_os(&format!("{}_{}", var_base, target_u)))
-                    .or_else(|| self.env_var_os(&format!("{}_{}", kind, var_base)))
+                    .or_else(|| self.env_var_os(&format!("{}_{}", var_base, kind_suffix)))
```

Comment:
This would now end up with `{var_base}_` if `host != target`. Note the trailing underscore.

This case should probably just not be tested at all under those conditions.

Reply:
```suggestion
                    .or_else(|| self.env_var_os(&format!("{}{}", var_base, kind_suffix)))
```
Ah yes, whoops, thanks